src/component.tsx

```diff
@@ -133,6 +135,7 @@ export function withRelay<Props extends RelayProps, ServerSideProps extends {}>(
 
       const nextPreloadedQuery = loadQuery(env, query, queryVariables, {
         fetchPolicy: 'store-or-network',
+        networkCacheConfig: opts.cacheConfigFromContext?.(router),
```

Comment:
How does this work when the identity of `router` changes? It won't re-render but it might capture stale values.

Reply:
Well its calculated on demand here when `queryVariables` change and I think thats correct.

For example router might change, but if queryVariables didn't change we don't want to do a new query. If `queryVariables` do change (for some reason), we will invoke this function here and the `router` will have the latest values.